source/tests/system/nirfmxnr_driver_api_tests.cpp

```diff
@@ -722,6 +715,98 @@ TEST_F(NiRFmxNRDriverApiTests, TxpContiguousMultiCarrierFromExample_FetchData_Da
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MEASUREMENT_TYPES_MODACC, false));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_SYNCHRONIZATION_MODE, NIRFMXNR_INT32_MODACC_SYNCHRONIZATION_MODE_FRAME));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_AVERAGING_ENABLED, NIRFMXNR_INT32_MODACC_AVERAGING_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_AVERAGING_COUNT, 10));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_MEASUREMENT_LENGTH_UNIT, NIRFMXNR_INT32_MODACC_MEASUREMENT_LENGTH_UNIT_SLOT));
+  EXPECT_SUCCESS(session, client::set_attribute_f64(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_MEASUREMENT_OFFSET, 0.0));
+  EXPECT_SUCCESS(session, client::set_attribute_f64(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_MEASUREMENT_LENGTH, 1));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_MAGNITUDE_AND_PHASE_ERROR_ENABLED, NIRFMXNR_INT32_MODACC_MAGNITUDE_AND_PHASE_ERROR_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_IQ_MISMATCH_ESTIMATION_ENABLED, NIRFMXNR_INT32_MODACC_IQ_MISMATCH_ESTIMATION_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_FREQUENCY_ERROR_ESTIMATION, NIRFMXNR_INT32_MODACC_FREQUENCY_ERROR_ESTIMATION_DISABLED));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_SYMBOL_CLOCK_ERROR_ESTIMATION_ENABLED, NIRFMXNR_INT32_MODACC_SYMBOL_CLOCK_ERROR_ESTIMATION_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_PHASE_TRACKING_ENABLED, NIRFMXNR_INT32_MODACC_PHASE_TRACKING_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_TIMING_TRACKING_ENABLED, NIRFMXNR_INT32_MODACC_TIMING_TRACKING_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_IQ_ORIGIN_OFFSET_ESTIMATION_ENABLED, NIRFMXNR_INT32_MODACC_IQ_ORIGIN_OFFSET_ESTIMATION_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_EVM_REFERENCE_DATA_SYMBOLS_MODE, NIRFMXNR_INT32_MODACC_EVM_REFERENCE_DATA_SYMBOLS_MODE_REFERENCE_WAVEFORM));
+
+  // READ TDMS File
+  auto waveform = load_test_waveform_data<float, nidevice_grpc::NIComplexNumberF32>("NR_FR2_UL_SISO_CC-1_BW-50MHz_SCS-120kHz.json");
+  // Split waveform data in two so it can fit within message limits.
+  std::vector<nidevice_grpc::NIComplexNumberF32> waveform_data_first_half(waveform.data.begin(), waveform.data.begin() + 300000);
```

Comment:
The NR waveform data from the TDMS file was rather large. It was above the message size limit configured for the server/client for the tests (I'm guessing the default). Rather than figure out how to adjust it I figured setting up the reference waveform twice with half the data each time would be good enough.